src/handler/ts6/chghost.rs

```diff
@@ -1,21 +1,21 @@
-use crate::handler::Outcome;
+use crate::handler::{Error, Outcome};
 use crate::line::Line;
 use crate::network::Network;
 use crate::util::DecodeHybrid as _;
 
 use super::TS6Handler;
 
 impl TS6Handler {
-    pub fn handle_chghost(network: &mut Network, line: &Line) -> Result<Outcome, &'static str> {
+    pub fn handle_chghost(network: &mut Network, line: &Line) -> Result<Outcome, Error> {
         if line.args.len() != 2 {
-            return Err("unexpected argument count");
+            return Err(Error::ExpectedArguments(2));
         }
 
         let uid = line.args[0].as_slice();
         let sid = &uid[..3];
 
         let server = network.servers.get_mut(sid).unwrap();
-        let user = server.users.get_mut(uid).ok_or("unknown uid")?;
+        let user = server.users.get_mut(uid).ok_or(Error::UnknownUser)?;
```

Comment:
This and a bunch of other lines that use `ok_or` also make me think it might be worth adding some methods to perform these lookups and return `Result`s with the correct `Error`s upon failure.